File: sieve_of_e.swift
// http://rosettacode.org/wiki/Sieve_of_Eratosthenes

func findPrimesTo(number:Int) -> [Int] {
  var primes = [Int](2...number)
  for var i = 0; i < primes.count; i++ {
    for var j = i+1; j < primes.count; j++ {
      if primes[j] % primes[i] == 0 {
        primes.removeAtIndex(j)
      }
    }
  }
  return primes
}

println(findPrimesTo(1000))

File: sieve_of_e.rb
# http://rosettacode.org/wiki/Sieve_of_Eratosthenes

def find_primes_below(limit)
  i = 0; p = (2..limit).to_a
  ( p.each { |n| p.delete(n) if n % p[i] == 0 && n != p[i] }; i += 1 ) until i >= p.count 
  return p
end

p find_primes_below(100)
